data/transactions/logic/eval_test.go

```diff
@@ -1034,7 +1034,7 @@ func TestGlobal(t *testing.T) {
 	addr, err := basics.UnmarshalChecksumAddress(testAddr)
 	require.NoError(t, err)
 	ledger.creatorAddr = addr
-	for v := uint64(0); v <= AssemblerMaxVersion; v++ {
+	for v := uint64(1); v <= AssemblerMaxVersion; v++ {
```

Comment:
why is it changed to 1? 0 is used to be a valid version for TEAL 1

Reply:
The trouble is that when `AssembleStringWithVersion` is given 0 as the version, the actual assembled version is 1, and so `#pragma version 1` is generated in the disassembly. But then `AssembleStringWithVersion(dissassembly, 0)` fails to reassemble it, because the source specifies 1 when the assembler was explicitly called with 0.  We could probably make all this work, but AssembleStringWithVersion is not actually used outside of testing, so it seems pointless to make it clever about accepting 1 when called with 0 (which might be reasonable!).  The only public use invokes it with assembleNoVersion. None of the other versioning tests test against 0, they all start at 1.

Reply:
I'm ok to have it set 1 here. The only problem this looks like the last place where assembling with version 0 happens. Probably a separate test checking assembling with version 0 at **ackwardCompat_test.go** is needed

Reply:
I don't see a good reason for `AssembleStringWithVersion(s, 0)` to work at all.  There's no "user-level" way to cause the function to run that way.  It always runs with the version `assemblerNoVersion` when invoked to assemble teal outside of tests.  I don't think we need to be compatible with old test code.  I don't think the WithVersion function should even be public.  And it should probably fail if given 0.